bitnami/grafana-alloy/templates/_helpers.tpl

```diff
@@ -0,0 +1,77 @@
+{{/*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{/*
+Return the proper Grafana Alloy image name
+*/}}
+{{- define "grafana-alloy.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.alloy.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper config reloader image name
+*/}}
+{{- define "config-reloader.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.configReloader.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the proper Docker Image Registry Secret Names
+*/}}
+{{- define "grafana-alloy.imagePullSecrets" -}}
+{{- include "common.images.renderPullSecrets" (dict "images" (list .Values.alloy.image .Values.configReloader.image) "context" $) -}}
+{{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "grafana-alloy.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "common.names.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Name of the Grafana Alloy ConfigMap
+*/}}
+{{- define "grafana-alloy.configmapName" -}}
+{{- if .Values.alloy.existingConfigmap -}}
+    {{- include "common.tplvalues.render" (dict "value" .Values.alloy.existingConfigmap "context" $) -}}
+{{- else -}}
+    {{- include "common.names.fullname" . -}}
+{{- end -}}
+{{- end -}}
+
+{{/* Validate values of Grafana Alloy - resource type */}}
+{{- define "grafana-alloy.validateValues.resourceType" -}}
+{{- if and (not (eq .Values.resourceType "daemonset")) (not (eq .Values.resourceType "statefulset")) (not (eq .Values.resourceType "deployment")) -}}
```

Comment:
One thing you can do to enhance reliability is to create an aux. variable to transform `resourceType` parameter to lowercase as we did in VictoriaMetrics, see:
- https://github.com/bitnami/charts/blob/main/bitnami/victoriametrics/templates/vmauth/dep-ds.yaml#L7
This way, users can install the chart using `helm install ... --set resourceType=StatefulSet` and it'll work.
```suggestion
{{- $resourceType := lower .Values.resourceType }} -}}
{{- if and (not (eq $resourceType "daemonset")) (not (eq $resourceType "statefulset")) (not (eq $resourceType "deployment")) -}}
```
If you apply this, ensure you do it on the rest of manifests where these comparisons are done.